scipy/stats/_continuous_distns.py

```diff
@@ -9153,18 +9153,25 @@ def _single_moment(K, k, df):
         return np.float64(ufunc(K, k, df))
 
     def _pdf(self, x, k, df):
-        cython_symbol = '_studentized_range_pdf'
 
         def _single_pdf(q, k, df):
-            log_const = _stats._studentized_range_pdf_logconst(k, df)
-            arg = [q, k, df, log_const]
-            usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+            # The infinite form of the PDF is derived from the infinite
+            # CDF.
+            if df < 100000:
```

Comment:
How did you choose the cutoff?

Reply:
That was selected way back when we investigating the performance of the log version of the CDF. It seemed that 100,000 was a safe limit for the standard integrand. Problems usually occurred at around 300,000. Here's an example of the graphs:
![k-19](https://user-images.githubusercontent.com/45153623/150611128-f657c6b8-131c-48d1-8a15-084fa760277e.png)
 
Now that you mention, it might not be valid for the PDF... I'll do some testing to see if it's OK.

Reply:
Yeah, that would be good to see. Probably not so different, but worth a look.

Reply:
Graphs are the PDF result vs df. 

The approximation switch over the entire range of df seems to work pretty well, and is barely even noticeable:
![q2-k3](https://user-images.githubusercontent.com/45153623/150653696-1f522e78-0515-4d3b-8f6a-3d3b64079923.png)

Zooming in on the switch region there's a clear transition, but it's not of huge consequence.
![q2 1-k3](https://user-images.githubusercontent.com/45153623/150653475-f9ed4466-b3a5-4a19-8412-de57cd6c9e7c.png)

The main point of the approximation switch is to handle np.inf inputs. Degrees of freedom above 120 aren't even tabulated in most tables, so situations where this transition actually matters are probably few and far between.

Reply:
> Graphs are the PDF result vs df.

At fixed `x` and `k`?

> so situations where this transition actually matters are probably few and far between.

I totally agree, but we should still have some rationale for our choice, or at least we should characterize the choice we make.

A simple way to do this would be to repeat our experiment (6.2 [here](https://zenodo.org/record/5151976#.Yex2nP7MKUk)) except with `df` fixed at 100000. Would that be pretty easy?

Reply:
Performed the experiment - After a bunch of testing seems like the integrator is bizarrely good at controlling error at high values of nu. Attached are plots of relative exact and approximate integrator error vs the reference, MP math implementation, as well as a plot of the "step" from the exact to approximate integrator.
![result](https://user-images.githubusercontent.com/45153623/157538905-4da6d8cf-d716-4500-8093-979a0db2c7fa.png)
![result_step](https://user-images.githubusercontent.com/45153623/157538925-1ce55542-ff58-4b6e-ac14-5c2d9521e711.png)

Below is a MPMath convergence plot to verify that the reference implementation (probably) isn't problematic. It's representative of all convergence plots preformed at a variety of Q and K values.
![2](https://user-images.githubusercontent.com/45153623/157539263-ebdd5457-12c9-4f45-bd64-b94c78af1777.png)

Reply:
to help other reviewers:

> Attached are plots of relative exact and approximate integrator error vs the reference, MP math implementation,

Is this the MP math implementation of the infinite `df` approximation or the MP math implementation for finite `df` evaluated at `df = 100000`?

Also, "SciPy Exact" means the implementation of `studentized_range.pdf` in `main`, and "SciPy Approx" means the infinite `df` approximation added by this PR, right?

Reply:
The MP math reference implementation is for the exact integral at `df=100000`, and your nomenclature is correct for exact/approx.